_posts/2024-11-26-opensearch-performance-2.17.md

```diff
@@ -0,0 +1,276 @@
+
+**Exact search improvements:** In OpenSearch 2.15, [SIMD optimizations](https://opensearch.org/blog/boosting-k-nn-exact-search/) were added to the k-NN plugin's script scoring, resulting in significant performance gains for CPUs with SIMD support, such as AVX2 or AVX512 on x86 or NEON on ARM. Further improvements in OpenSearch 2.17 introduced Lucene's new vector format, which includes optimized memory-mapped file access. Together, these enhancements significantly reduce search latency for exact k-NN searches on supported hardware.
+
+# Roadmap for 2025
+
+The following improvements are included in the OpenSearch roadmap for 2025.
+
+## Core search engine
+
+In 2025, we will push the boundaries of OpenSearch's query engine with several key initiatives aimed at improving performance, scalability, and efficiency:
+
+* **[Streaming architecture](https://github.com/opensearch-project/OpenSearch/issues/16679)**: We're moving from a request/response model to streaming, processing and delivering data in real time, reducing memory overhead and improving query speed.
+* **[Native join support](https://github.com/opensearch-project/OpenSearch/issues/15185)**: We're introducing efficient JOIN operations across indexes to be supported natively, fully integrated with OpenSearch's query DSL, PPL, and SQL.
+* **Native vectorized processing**: By using modern CPU SIMD operations and native code, we're optimizing the processing of data streams to eliminate Java's garbage collection bottlenecks.
+* **[Smarter query planning](https://github.com/opensearch-project/OpenSearch/issues/12390)**: Optimizing where and how computations run ensures we reduce unnecessary data transfer and improve performance for parallel query execution.
+* **[gRPC-based Search API](https://github.com/opensearch-project/OpenSearch/issues/15190)**: We're enhancing client-server communication with Protobuf and gRPC, accelerating search by reducing overhead.
+* **[Query performance optimization](https://github.com/orgs/opensearch-project/projects/153)**: Improving performance remains our priority, and several key initiatives such as docId encoding and query approximation will reduce index size and enhance performance of large-range queries.
+* **[Star-Tree indexing](https://github.com/opensearch-project/OpenSearch/issues/12498)**: Precomputing aggregations using Star-Tree indexing ensures faster, more predictable performance for aggregation-heavy queries.
+
+### Vector search
+
+In 2025, we will continue to invest in the following key initiatives aimed at performance improvements and cost savings:
+
+* **Index build acceleration with GPUs and SIMD:** k-NN performance can be enhanced by using libraries with GPU support. Because vector distance calculations are compute-heavy, GPUs can speed up computations and enhance the performance of index build time and search queries.
```

Comment:
```suggestion
* **Index build acceleration with GPUs and SIMD:** k-NN performance can be enhanced by using libraries with GPU support. Because vector distance calculations are compute-heavy, GPUs can speed up computations and reduce index build and search query times.
```